src/Middleware/CORS/src/Infrastructure/CorsMiddleware.cs

```diff
@@ -145,8 +145,23 @@ private async Task InvokeCore(HttpContext context, ICorsPolicyProvider corsPolic
             // Get the most significant CORS metadata for the endpoint
             // For backwards compatibility reasons this is then downcast to Enable/Disable metadata
             var corsMetadata = endpoint?.Metadata.GetMetadata<ICorsMetadata>();
+
             if (corsMetadata is IDisableCorsAttribute)
             {
+                var isOptionsRequest = string.Equals(
+                    context.Request.Method,
+                    CorsConstants.PreflightHttpMethod,
+                    StringComparison.OrdinalIgnoreCase);
+
+                var isCorsPreflightRequest = isOptionsRequest && context.Request.Headers.ContainsKey(CorsConstants.AccessControlRequestMethod);
+
+                if (isCorsPreflightRequest)
+                {
+                    // If this is a preflight request, and we disallow CORS, complete the request
+                    context.Response.StatusCode = StatusCodes.Status204NoContent;
+                    return;
+                }
+
```

Comment:
That's kinda interesting that we missed this in the middleware. so were preflight requests going through to the the thing that handles the request? I'm not 100% certain of the ramifications of this change.

Reply:
We don't have any tests outside of MVC that use `IDisableCorsAttribute`. It's only meaningful (not really) when you `AllowCors` on controllers and turn it off for an action. Prior to this change, the filter would have done the needful, at least with MVC.